lapis2/src/test/kotlin/org/genspectrum/lapis/controller/LapisControllerCsvTest.kt

```diff
@@ -30,547 +35,412 @@ class LapisControllerCsvTest(@Autowired val mockMvc: MockMvc) {
+        if (endpoint == "/details") {
+            every {
+                siloQueryModelMock.getDetails(sequenceFiltersRequestWithFields(mapOf("country" to "Switzerland")))
+            } returns emptyList()
+        }
+        if (endpoint == "/aggregated") {
+            every {
+                siloQueryModelMock.getAggregated(sequenceFiltersRequestWithFields(mapOf("country" to "Switzerland")))
+            } returns emptyList()
+        }
+        if (endpoint == "/nucleotideMutations") {
+            every { siloQueryModelMock.computeNucleotideMutationProportions(any()) } returns emptyList()
+        }
+        if (endpoint == "/aminoAcidMutations") {
+            every { siloQueryModelMock.computeAminoAcidMutationProportions(any()) } returns emptyList()
+        }
+        if (endpoint == "/nucleotideInsertions") {
+            every { siloQueryModelMock.getNucleotideInsertions(any()) } returns emptyList()
+        }
+        if (endpoint == "/aminoAcidInsertions") {
+            every { siloQueryModelMock.getAminoAcidInsertions(any()) } returns emptyList()
+        }
+    }
+
+    fun mockEndpointReturnData(endpoint: String) {
+        if (endpoint == "/details") {
+            every {
+                siloQueryModelMock.getDetails(sequenceFiltersRequestWithFields(mapOf("country" to "Switzerland")))
+            } returns detailsData
+        }
+        if (endpoint == "/aggregated") {
+            every {
+                siloQueryModelMock.getAggregated(sequenceFiltersRequestWithFields(mapOf("country" to "Switzerland")))
+            } returns aggregationData
+        }
+        if (endpoint == "/nucleotideMutations") {
+            every { siloQueryModelMock.computeNucleotideMutationProportions(any()) } returns nucleotideMutationData
+        }
+        if (endpoint == "/aminoAcidMutations") {
+            every { siloQueryModelMock.computeAminoAcidMutationProportions(any()) } returns aminoAcidMutationData
+        }
+        if (endpoint == "/nucleotideInsertions") {
+            every { siloQueryModelMock.getNucleotideInsertions(any()) } returns nucleotideInsertionData
+        }
+        if (endpoint == "/aminoAcidInsertions") {
+            every { siloQueryModelMock.getAminoAcidInsertions(any()) } returns aminoAcidInsertionData
+        }
+    }
+
+    fun returnedCsvData(endpoint: String): String {
+        if (endpoint == "/details") {
+            return detailsDataCsv
+        }
+        if (endpoint == "/aggregated") {
+            return aggregationDataCsv
+        }
+        if (endpoint == "/nucleotideMutations") {
+            return mutationDataCsv
+        }
+        if (endpoint == "/aminoAcidMutations") {
+            return mutationDataCsv
+        }
+        if (endpoint == "/nucleotideInsertions") {
+            return nucleotideInsertionDataCsv
+        }
+        if (endpoint == "/aminoAcidInsertions") {
+            return aminoAcidInsertionDataCsv
+        }
+        return ""
+    }
```

Comment:
`when` could make this a lot more concise:
```suggestion
    fun returnedCsvData(endpoint: String) = when (endpoint) {
        "/details" -> detailsDataCsv
        "/aggregated" -> aggregationDataCsv
        "/nucleotideMutations" -> mutationDataCsv
        "/aminoAcidMutations" -> mutationDataCsv
        "/nucleotideInsertions" -> nucleotideInsertionDataCsv
        "/aminoAcidInsertions" -> aminoAcidInsertionDataCsv
        else -> throw ...
    }
```